### ISSUE TITLE
B031 does not take into account if-else statements

### ISSUE THREAD

**Issue by amyasnikov** [PyCQA/flake8-bugbear#465]
Bugbear reports false-positive "more that one" usage of the generator even if it occurs inside 2 different branches of the if-else statement.

This code
```
import itertools

for _, group in itertools.groupby([]):
    if True:
        print(group, True)
    else:
        print(group, False)
```

produces
```
b031.py:7:15: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.
```
but it's obvious that `group` cannot be used twice here.

flake8 --version:
```
7.0.0 (flake8-bugbear: 24.2.6, mccabe: 0.7.0, pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.11.2 on Linux
```

**Comment by cooperlees** [comment]
Thanks for reporting. Both are things we could tighten. PRs welcome.

**Comment by Sonike** [comment]
I reproduced this on the current `main` branch and would like to work on the if/else false positive. I plan to make B031 count usages per mutually exclusive control-flow path, while preserving warnings for repeated uses within the same branch and for uses before or after the conditional. I'll add focused eval fixtures and run the full test/lint suite.

I'll keep this change scoped to the if/elif/else behavior; the annotation case mentioned in the comments has already been addressed separately.
